### PR TITLE
CLI: Add url flag to download templates in sb repro

### PR DESCRIPTION
**Pull request by yannbf**
Issue: N/A

## What I did

This PR integrates `degit` into the sb cli.

It adds two features:

### Prefer downloading from storybook repro repository

This way, the CLI just downloads a git repo from [repro-templates](https://github.com/storybookjs/repro-templates) instead of having to run a generator (e.g. create-react-app myApp...). This could save time both in CI but also for the `sb repro` experience.

If the selected template does not exist in repro-templates, it will fallback to the original generator.

For the user, there's no change in the command. Optionally, you could use the `--skip-git` flag to use the generators directly instead.

The repro template has both `main` and `next` branches which old latest vs next versions of Storybook. The script will detect whether the user is using `sb repro` or `sb repro@some-alpha-version` and will download from the correct branch

### Set project via --url flag

`sb repro` now has a `--url` flag which allows you to download any git project (github, gitlab, bitbucket..) and bootstrap Storybook on top of it. Could be useful for QAing purposes in case we find big projects with certain settings that could be valuable to test out. 

This uses `degit` under the hood so you can use all kinds of formats for the URL such as `username/project` or `https://github.com/username/project` or other formats that degit supports.

## How to test

First check the branch, bootstrap it or just build the cli:
```
yarn build cli
```

Next, somewhere in an experimentation folder, run `sb repro` using the URL of a project that does not contain Storybook.

```
# This will prompt for location folder, using the repo as default folder name
../storybook/lib/cli/bin/index.js repro --url the-url-here
```

```
# This will download and install things automatically with no prompt
../storybook/lib/cli/bin/index.js repro --url the-url-here outputfoldername
```

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: lib/cli/src/repro-generators/configs.ts [hunk 1]
```diff
@@ -17,6 +17,10 @@ export interface Parameters {
   typescript?: boolean;
   /** Merge configurations to main.js before running the tests */
   mainOverrides?: Partial<StorybookConfig> & Record<string, any>;
+  /** Generator that downloads from remote repro in storybookjs/repro-templates on Github */
+  gitRepoGenerator?: string;
+  /** Whether to download from git templates */
+  preferGitTemplate?: boolean;
```
Comment (shilman) [inline lib/cli/src/repro-generators/configs.ts:23]: I think this shouldn't be included in the code. It should just try & fail & run the full generator OR the user should pass `--skip-git`.

**Review by shilman — CHANGES_REQUESTED** [review]
Looks like this breaks CI. Can you please take a look when you get back from holiday?

File: lib/cli/src/repro-generators/configs.ts (hunk 1 above)
Comment (yannbf) [reply lib/cli/src/repro-generators/configs.ts:23]: That's the case. I just added that name to make it more apparent, but that flag is set based on --skip-git

**Comment by yannbf** [comment]
> Looks like this breaks CI. Can you please take a look when you get back from holiday?

If we want to already have CI download repos we will need to fix the repro-generators script first

**Comment by yannbf** [comment]
Also @shilman here's a small summary of timing changes from before/after. This will decrease a ton as we increase the amount of repros in the repro-templates repository

image

**Comment by yannbf** [comment]
@shilman in order for this to work with `future` we'll need a `future` branch in the repro templates. Else this could lead to a situation where people do `npx sb@future repro` and it will actually download from `repro-templates` at `next` branch

**Comment by yannbf** [comment]
This PR borked because of a bad merge using future/base rather than next.

Closing it in favor of https://github.com/storybookjs/storybook/pull/18795

**Pull request closed without merging**